app/models/field.rb

```diff
@@ -0,0 +1,17 @@
+class Field < ApplicationRecord
+  belongs_to :calculator
+  before_validation :set_uuid, :set_selector
+  validates :uuid, :type, :label, :kind, presence: true
+  enum kind: { form: 0, parameter: 1, result: 2 }
+
+  private
+    def set_uuid
+      self.uuid = SecureRandom.uuid if self.uuid.nil?
+    end
+    
+    def set_selector
+      existed_selectors_count = Field.where(type: :kind).count
```

Comment:
`count` may lead to duplication. For example, you've created fields with selectors: F1, F2, F3. And then removed the field with selector F2. The new field will be created with F3 again

Reply:
Thank you for the explanation! So as I've understood it's better to take into account the number of the last selector when creating the new.